### PR TITLE
return model id in registering remote model

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
This change only applies to registering remote models. Registering a remote model doesn't need Asynchronize, it should return model_id right away in the response. 

**Before this change:**
POST /_plugins/_ml/models/_register
{
    "name": "openAI-GPT-3.5 completions",
    "function_name": "remote",
    "description": "test model",
    "connector_id": "kUFFL4kB4ubqQRzeQ_r-"
}
Response:
{
  "task_id": "l0FIL4kB4ubqQRzeKvpV",
  "status": "CREATED"
}

**After this change:**
POST /_plugins/_ml/models/_register
{
    "name": "openAI-GPT-3.5 completions",
    "function_name": "remote",
    "description": "test model",
    "connector_id": "kUFFL4kB4ubqQRzeQ_r-"
}
Response:
{
  "task_id": "l0FIL4kB4ubqQRzeKvpV",
  "status": "CREATED",
  "model_id": "xxxxxxxxxxxxxx",
}

### CODE REVIEW

**Review by b4sjoo — APPROVED** [review]

**Review by rbhavna — APPROVED** [review]

File: common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelResponse.java [hunk 1]
```diff
@@ -18,33 +18,46 @@
...
     private String taskId;
     private String status;
+    private String modelId;
 
     public MLRegisterModelResponse(StreamInput in) throws IOException {
         super(in);
         this.taskId = in.readString();
         this.status = in.readString();
+        this.modelId = in.readOptionalString();
     }
```
Comment (saratvemulapalli) [inline common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelResponse.java:33]: Feels like a good improvement for remote models, do we really have a taskId which is doing real work for remote models?

File: plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java [hunk 2]
```diff
@@ -249,8 +250,8 @@ private void registerModel(MLRegisterModelInput registerModelInput, ActionListen
             mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
                 String taskId = response.getId();
                 mlTask.setTaskId(taskId);
-                mlModelManager.registerMLModel(registerModelInput, mlTask);
-                listener.onResponse(new MLRegisterModelResponse(taskId, MLTaskState.CREATED.name()));
+                System.out.println("mlModelManager calls registerMLRemoteModel");
```
Comment (saratvemulapalli) [inline plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:253]: Nit: Probably unintended?

Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:253]: Are we keeping this line?

File: plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java [hunk 3]
```diff
@@ -233,6 +233,7 @@ private void registerModel(MLRegisterModelInput registerModelInput, ActionListen
                 throw new IllegalArgumentException("URL can't match trusted url regex");
             }
         }
+        System.out.println("registering the model");
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:236]: Are we keeping this?

Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:236]: Let's avoid using system out print

File: plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java [hunk 4]
```diff
@@ -388,6 +450,95 @@ public void registerMLModel(MLRegisterModelInput registerModelInput, MLTask mlTa
...
+            .doc(modelGroupSourceMap);
+
+        return updateModelGroupRequest;
+    }
+
+    private int incrementLatestVersion(Map<String, Object> modelGroupSourceMap) {
+        return (int) modelGroupSourceMap.get(MLModelGroup.LATEST_VERSION_FIELD) + 1;
+    }
+
+    private void indexRemoteModel(
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java:479]: Do we need to the other method: `private void indexRemoteModel(MLRegisterModelInput registerModelInput, MLTask mlTask, String modelVersion)` as we are invoking this method only.

File: plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java [hunk 5]
```diff
@@ -222,30 +224,22 @@ public void registerModelMeta(MLRegisterModelMetaInput mlRegisterModelMetaInput,
...
+                            UpdateRequest updateModelGroupRequest = createUpdateModelGroupRequest(
+                                modelGroupSource,
+                                modelGroupId,
+                                modelGroup.getSeqNo(),
+                                modelGroup.getPrimaryTerm(),
+                                updatedVersion
+                            );
+
+                            client.update(updateModelGroupRequest, ActionListener.wrap(r -> {
+                                uploadMLModelMeta(mlRegisterModelMetaInput, updatedVersion + "", listener);
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java:238]: regarding : `updatedVersion + ""` is the goal is to convert int to string, may be we can follow standard way like: `Integer.toString(updatedVersion)`?

File: plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java (hunk 2 above)
Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:253]: Good catch! I think I forgot to delete. They will be cleaned up for sure.

File: plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java [hunk 6]
```diff
@@ -249,8 +250,8 @@ private void registerModel(MLRegisterModelInput registerModelInput, ActionListen
             mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
                 String taskId = response.getId();
                 mlTask.setTaskId(taskId);
-                mlModelManager.registerMLModel(registerModelInput, mlTask);
-                listener.onResponse(new MLRegisterModelResponse(taskId, MLTaskState.CREATED.name()));
+                System.out.println("mlModelManager calls registerMLRemoteModel");
+                mlModelManager.registerMLRemoteModel(registerModelInput, mlTask, listener);
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:254]: Is this the expectation that, only remote model registration will be asynchronous? If yes, can we please add a comment here also?

File: plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java (hunk 4 above)
Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java:479]: Yes it's needed for registering non-remote models. Actually if you use the other method, the model_id will not be included and the model registering could be asynchronize.

File: plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java (hunk 3 above)
Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:236]: Yes I guarante these redundant lines will be removed in my next PR.

File: plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java (hunk 6 above)
Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/action/register/TransportRegisterModelAction.java:254]: Just the opposite, remote model registration is synchronous only so the model_id can be returned right away. Comments will be added in the next PR.

File: common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelResponse.java (hunk 1 above)
Comment (saratvemulapalli) [reply common/src/main/java/org/opensearch/ml/common/transport/register/MLRegisterModelResponse.java:33]: Curious what kind of validations ?

**Pull request merged**